### PR TITLE
Add recipes for trio and async_generator

### PR DESCRIPTION
**Pull request by nicoddemus**

### CODE REVIEW

**Comment by nicoddemus** [comment]
@conda-forge/staged-recipes I believe this is ready for review/merge. 👍

File: recipes/async_generator/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    - async_generator
+    - async_generator._tests
+
+about:
+  home: https://github.com/python-trio/async_generator
+  license: MIT
+  license_family: MIT
+  license_file: ''
+  summary: Async generators and context managers for Python 3.5+
+  description: ".. image:: https://img.shields.io/badge/chat-join%20now-blue.svg\n   :target: https://gitter.im/python-trio/general\n   :alt: Join chatroom\n\n.. image:: https://img.shields.io/badge/docs-read%20now-blue.svg\n\
```
Comment (isuruf) [inline recipes/async_generator/meta.yaml]: This is unreadable. I would just take a paragraph or 2 and avoid the examples

**Comment by nicoddemus** [comment]
I agree, this was generated by conda build. Will change it later.

Em sáb, 3 de fev de 2018 13:19, Isuru Fernando <notifications@github.com>
escreveu:

File: recipes/async_generator/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+test:
+  imports:
+    - async_generator
+    - async_generator._tests
+
+about:
+  home: https://github.com/python-trio/async_generator
+  license: MIT
+  license_family: MIT
+  license_file: ''
```
Comment (CJ-Wright) [inline recipes/async_generator/meta.yaml]: Can we get a license?

File: recipes/async_generator/meta.yaml (hunk 1 above)
Comment (nicoddemus) [reply recipes/async_generator/meta.yaml]: Fixed, thanks

File: recipes/async_generator/meta.yaml (hunk 2 above)
Comment (nicoddemus) [reply recipes/async_generator/meta.yaml]: Opened python-trio/async_generator#11 and added a link to the recipe so we can fix that in the next release.

**Comment by nicoddemus** [comment]
@CJ-Wright @isuruf ready for another review.

**Comment by nicoddemus** [comment]
cc @conda-forge/staged-recipes

File: recipes/async_generator/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+{% set version = "1.9" %}
+{% set hash_type = "sha256" %}
+{% set hash_value = "b7d5465c6174fe86dba498ececb175f93a6097ffb7cc91946405e1f05b848371" %}
+
+package:
+  name: '{{ name|lower }}'
+  version: '{{ version }}'
+
+source:
+  fn: '{{ name }}-{{ version }}.tar.gz'
```
Comment (ocefpaf) [inline recipes/async_generator/meta.yaml]: This line can be dropped.

File: recipes/async_generator/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  '{{ hash_type }}': '{{ hash_value }}'
+
+build:
+  number: 0
+  script: python setup.py install  --single-version-externally-managed --record=record.txt
```
Comment (ocefpaf) [inline recipes/async_generator/meta.yaml]: We are moving to ``script: python -m pip install --no-deps --ignore-installed .``. See https://github.com/conda-forge/conda-forge.github.io/pull/518

File: recipes/async_generator/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  imports:
+    - async_generator
+    - async_generator._tests
+
+about:
+  home: https://github.com/python-trio/async_generator
+  license: MIT
+  license_family: MIT
+  # license file not available on sdist yet: https://github.com/python-trio/async_generator/pull/11
+  license_file: ''
```
Comment (ocefpaf) [inline recipes/async_generator/meta.yaml]: You don't need the empty field but that does not hurt.

File: recipes/trio/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    - trio.testing
+    - trio.tests
+
+about:
+  home: https://github.com/python-trio/trio
+  license: MIT
+  license_family: MIT
+  license_file: LICENSE
+  summary: An async/await-native I/O library for humans and snake people
+  description: ".async/await-native I/O library for Python. Like all async libraries,\n
```
Comment (ocefpaf) [inline recipes/trio/meta.yaml]: Kind of big and with some escape characters that I don't think are needed.

File: recipes/async_generator/meta.yaml (hunk 4 above)
Comment (nicoddemus) [reply recipes/async_generator/meta.yaml]: Done

File: recipes/async_generator/meta.yaml (hunk 3 above)
Comment (nicoddemus) [reply recipes/async_generator/meta.yaml]: Done

File: recipes/trio/meta.yaml (hunk 6 above)
Comment (nicoddemus) [reply recipes/trio/meta.yaml]: Done

File: recipes/async_generator/meta.yaml (hunk 5 above)
Comment (nicoddemus) [reply recipes/async_generator/meta.yaml]: I prefer to leave it there so we fill it with the license file in the next release

**Comment by ocefpaf** [comment]
@nicoddemus What do you want to do about the AppVeyor failure? `async_generator` is noit building on py27:

```
Processing c:\bld\async_generator_1517761572846\work\async_generator-1.9
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "c:\users\appveyor\appdata\local\temp\1\pip-msxyjb-build\setup.py", line 1, in <module>
        from pathlib import Path
    ImportError: No module named pathlib
    
    ----------------------------------------
```

**Comment by nicoddemus** [comment]
`async_generator` and `trio` are py35+ only, so they are not supposed to be built in py27. `async_generator` is `noarch: python`, which is why it fails to build on py27.

Should I make `async_generator` **not** `noarch: python` so it doesn't try to build on py27?

**Comment by nicoddemus** [comment]
> You can leave the noarch there and add a constraint like python >2.7 for build and run.

Ahh good idea, done!

**Comment by nicoddemus** [comment]
Thanks @ocefpaf!

**Pull request merged**